### PR TITLE
Onboarding v1

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot/pull/2172
image*
image
image
image
image
image

### CODE REVIEW

File: profiles/daily_trading/profile.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "profile": {
         "avatar": "default_profile.png",
         "description": "DailyTrading is a low risk versatile profile that catches buy and sell opportunities using RSI and moving average indicators.",
-        "id": "default",
+        "id": "daily_trading",
```
Comment (Herklos) [inline profiles/daily_trading/profile.json:39]: So blank will be the default?

**Review by Herklos — APPROVED** [review]

File: profiles/daily_trading/profile.json (hunk 1 above)
Comment (GuillaumeDSM) [reply profiles/daily_trading/profile.json:39]: yes, and onboarding will enable the user to change it

Comment (Herklos) [reply profiles/daily_trading/profile.json:39]: :+1:

File: Services/Interfaces/web_interface/templates/components/community/login.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            <a class="btn btn-sm btn-outline-warning waves-effect mt-4" href="{{ url_for('community_logout') }}">Logout</a>
+        {% else %}
+            <h5>
+                Login and access your
+                {% if is_in_stating_community_env() %}
+                    <span class="badge badge-light">
+                        Beta
+                    </span>
+                {% endif %}
+                Astrolab account.
```
Comment (Herklos) [inline Services/Interfaces/web_interface/templates/components/community/login.html:31]: I think we will need to rename Astrolab mentions :smile:

**Review by Herklos — CHANGES_REQUESTED** [review]

File: Services/Interfaces/web_interface/api/metadata.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
 def ping():
     start_time = interfaces.get_bot_api().get_start_time()
     return json.dumps(
-        f"Running sice {timestamp_util.convert_timestamp_to_datetime(start_time, '%Y-%m-%d %H:%M:%S')}."
+        f"Running since {timestamp_util.convert_timestamp_to_datetime(start_time, '%Y-%m-%d %H:%M:%S')}."
```
Comment (Herklos) [inline Services/Interfaces/web_interface/api/metadata.py:31]: :smile:

File: Services/Interfaces/web_interface/controllers/onboarding.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import tentacles.Services.Interfaces.web_interface as web_interface
+import tentacles.Services.Interfaces.web_interface.login as login
+import tentacles.Services.Interfaces.web_interface.models as models
+import tentacles.Services.Interfaces.web_interface.controllers.community_authentication as community_authentication
+import octobot_services.interfaces.util as interfaces_util
+
+
+@web_interface.server_instance.route("/onboarding")
+@login.login_required_when_activated
+def onboarding():
```
Comment (Herklos) [inline Services/Interfaces/web_interface/controllers/onboarding.py:29]: :tada:

File: Services/Interfaces/web_interface/static/js/components/onboarding.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    const ensure_modales = () => {
+        $('button[data-toggle="modal"]').each((_, element) => {
+            $(element).click((jsElement) => {
+                const element = $(jsElement.target);
+                element.parent().children(element.data("target")).modal();
+            })
+        })
+    }
+
+    ensure_modales();
```
Comment (Herklos) [inline Services/Interfaces/web_interface/static/js/components/onboarding.js:31]:
```suggestion
    ensure_modals();
```

File: Trading/Mode/blank_trading_mode/blank_trading.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+import octobot_trading.enums as enums
+import octobot_trading.modes as trading_modes
+
+
+class BlankTradingMode(trading_modes.AbstractTradingMode):
```
Comment (Herklos) [inline Trading/Mode/blank_trading_mode/blank_trading.py:21]: :+1:

**Review by Herklos — COMMENTED** [review]
Should we update default profile png image with the new OctoBot logo ?

File: Services/Interfaces/web_interface/controllers/community_authentication.py [hunk 7]
```diff
@@ -45,7 +47,12 @@ def community_login():
                     log_exceptions=False
                 )
                 logged_in_email = form.email.data
-                return flask.redirect('community')
+                if after_login_action == "sync_account":
+                    added_profiles = models.sync_community_account()
+                    if added_profiles:
+                        flask.flash(f"Downloaded {len(added_profiles)} profile{'s' if len(added_profiles) > 1 else ''} "
```
Comment (Herklos) [inline Services/Interfaces/web_interface/controllers/community_authentication.py:53]: It is displayed to the user?

File: Services/Interfaces/web_interface/models/configuration.py [hunk 8]
```diff
@@ -638,12 +638,16 @@ def get_symbol_list(exchanges):
     return list(set(result))
 
 
+def _get_filtered_exchange_symbols(symbols):
+    return [res for res in symbols if "/" in res]
```
Comment (Herklos) [inline Services/Interfaces/web_interface/models/configuration.py:642]: Shouldn't we use symbol_utils ?

File: Services/Interfaces/web_interface/templates/onboarding.html [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+                   href="{{ url_for('community_logout', next=url_for('onboarding'))}}">
+                    logout
+                </a>
+            </div>
+            {% endif %}
+        </div>
+        {{ m_flash_messages.flash_messages() }}
+        {% else %}
+        <div>
+            Login to you OctoBot Account to access your subscribed profiles as well as the additional profiles.
```
Comment (Herklos) [inline Services/Interfaces/web_interface/templates/onboarding.html:99]:
```suggestion
            Login to your OctoBot Account to access your subscribed profiles as well as the additional profiles.
```

File: Services/Interfaces/web_interface/templates/onboarding.html [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+            <div class="login_box mx-auto mt-1 mt-xl-5">
+                {{ login.login_form(form, is_in_stating_community_env, url_for("onboarding"), "sync_account") }}
+            </div>
+        </div>
+        {% endif %}
+        <h3>
+            Select the profile your OctoBot use
```
Comment (Herklos) [inline Services/Interfaces/web_interface/templates/onboarding.html:107]: Should we hide this part while the user is not connected?

**Review by Herklos — COMMENTED** [review]
Great onboarding page! It will greatly simplify user setup :100:

File: Services/Interfaces/web_interface/controllers/configuration.py [hunk 11]
```diff
@@ -36,12 +36,19 @@
 def profile():
     selected_profile = flask.request.args.get("select", None)
     next_url = flask.request.args.get("next", None)
+    reboot = flask.request.args.get("reboot", False)
+    reboot_delay = 3
     if selected_profile is not None and selected_profile != models.get_current_profile().profile_id:
         models.select_profile(selected_profile)
         current_profile = models.get_current_profile()
-        flask.flash(f"Switched to {current_profile.name} profile.", "success")
+        if not reboot:
```
Comment (Herklos) [inline Services/Interfaces/web_interface/controllers/configuration.py:44]: Should we flash something before rebooting like "Applying configuration..." ?

File: Services/Interfaces/web_interface/controllers/home.py [hunk 12]
```diff
@@ -25,17 +25,13 @@
 @web_interface.server_instance.route("/home")
 @login.login_required_when_activated
 def home():
-    first_display = False
-    if flask.request.args:
-        accepted = flask.request.args.get("accept_terms") == "True"
-        models.accept_terms(accepted)
-        first_display = True
+    display_tutorial = str(flask.request.args.get("display_tutorial", False)) == "True"
```
Comment (Herklos) [inline Services/Interfaces/web_interface/controllers/home.py:28]: Do you think we could use something like [this](https://github.com/usablica/intro.js) for the tutorial ?

File: Services/Interfaces/web_interface/models/commands.py [hunk 13]
```diff
@@ -26,10 +29,20 @@ def _delayed_command():
...
+    _PENDING_COMMANDS.add(_REBOOT)
+    if delay:
+        # recall self with delay
+        schedule_delayed_command(restart_bot, delay=delay)
+        return
+    _PENDING_COMMANDS.remove(_REBOOT)
     interfaces_util.get_bot_api().restart_bot()
 
 
+def is_rebooting():
```
Comment (Herklos) [inline Services/Interfaces/web_interface/models/commands.py:42]: :+1:

**Review by Herklos — COMMENTED** [review]
:+1:

File: Services/Interfaces/web_interface/controllers/community_authentication.py (hunk 7 above)
Comment (GuillaumeDSM) [reply Services/Interfaces/web_interface/controllers/community_authentication.py:53]: yes, as here https://user-images.githubusercontent.com/9078616/209992555-ec9ebcad-1550-4894-a38f-596c275947b7.png

File: Services/Interfaces/web_interface/templates/onboarding.html (hunk 10 above)
Comment (GuillaumeDSM) [reply Services/Interfaces/web_interface/templates/onboarding.html]: if we want to force connection on self hosted bots then we should, for now connecting is making subscribed profiles available on self hosted bots. I think it's better to make it possible not to login on self hosted bots

File: Services/Interfaces/web_interface/controllers/configuration.py (hunk 11 above)
Comment (GuillaumeDSM) [reply Services/Interfaces/web_interface/controllers/configuration.py:44]: there is a waiter card that appears when rebooting

File: Services/Interfaces/web_interface/flask_util/browsing_data_provider.py [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+    def set_first_displays(self, is_first_display):
+        for key in self.browsing_data[self.FIRST_DISPLAY]:
+            self.browsing_data[self.FIRST_DISPLAY][key] = is_first_display
+        self._dump_saved_data()
+
+    def _get_session_secret_key(self):
+        return commons_configuration.decrypt(self.browsing_data[self.SESSION_SEC_KEY]).encode()
+
+    def _create_session_secret_key(self):
+        # always generate a new unique session secret key, reuse it to save sessions after restart
```
Comment (Herklos) [inline Services/Interfaces/web_interface/flask_util/browsing_data_provider.py:66]: 💯

File: Services/Interfaces/web_interface/login/web_login_manager.py [hunk 15]
```diff
@@ -32,13 +32,20 @@
 
 class WebLoginManager(flask_login.LoginManager):
     def __init__(self, flask_app, password_hash):
+        # force is_authenticated to save login state throughout server restart
```
Comment (Herklos) [inline Services/Interfaces/web_interface/login/web_login_manager.py:35]: 👍

File: Services/Interfaces/web_interface/static/js/common/tutorial.js [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+
+function getDocsLink(route, name) {
+    return `<a class="" target="_blank" rel="noopener" href="${getDocsUrl()}${route}">${name}</a>`
+}
+
+function getExchangesDocsLink(route, name) {
+    return `<a class="" target="_blank" rel="noopener" href="${getExchangesDocsUrl()}${route}">${name}</a>`
+}
+
+_TUTORIALS = {
```
Comment (Herklos) [inline Services/Interfaces/web_interface/static/js/common/tutorial.js:13]: 🎉

File: Services/Interfaces/web_interface/static/js/common/tutorial.js [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+                },
+                {
+                    title: "That's it !",
+                    intro: 'We hope you will enjoy OctoBot. Use the <a class="blue-text" target="_blank"><i class="fa-solid fa-question"></i></a> buttons to learn more on how to use OctoBot'
+                },
+            ]
+        }
+    },
+
+    profile: () => {
```
Comment (Herklos) [inline Services/Interfaces/web_interface/static/js/common/tutorial.js:58]: 👍

**Review by Herklos — COMMENTED** [review]
👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**